### PR TITLE
Add authentication documentation

### PR DESCRIPTION
**Pull request by tschuy**

### CODE REVIEW

**Comment by Kennric** [comment]
I realize this is a draft, but I'd like to be more definite in our api docs - that's a general description of how we do auth sections in the api, but I'd like a concrete spec for "this is exactly what to send specifically for password authentication" - maybe make the example there more general and have a separate section for the password auth specifics

File: source/draft_auth.rst [hunk 1]
```diff
@@ -3,3 +3,49 @@
...
+An auth block typically looks like the above example, with a ``type`` field
+specifying what kind of authentication the client is connecting with, an other
+fields as necessary for that kind of connection.
+
+These other fields might be things like ``password``, ``token``, ``key``, etc.
+
+Password authentication
+-----------------------
+
+When used with password-based authentication, TimeSync uses a username field and
```
Comment (Kennric) [inline source/draft_auth.rst]: I would say "requires" instead of "uses" in this sentence, otherwise this is good

**Comment by Kennric** [comment]
I like it. +1

**Comment by pop** [comment]
I also like it. Simple and easy to understand. +1

**Pull request merged**